gauss_plot_2d.ipynb

```diff
@@ -0,0 +1,297 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #20.    gauss_pdf = jnp.vectorize(multivariate_pdfs.multi_normal, excluded=[2])
</pre><p>Use <code>jax.vmap()</code>  instead of <code>jnp.vectorize()</code> as suggested by Dr murphy.  Check on jax docs how jax.vmap() works.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Reply:
[jax.numpy.vectorize()](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.vectorize.html#jax.numpy.vectorize) has the same interface as [numpy.vectorize](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html#numpy.vectorize), but it is syntactic sugar for an auto-batching transformation (vmap()) rather than a Python loop. This should be considerably more efficient, but the implementation must be written in terms of functions that act on JAX arrays.


I have removed the class and directly using the function for vectorization. Do I need to change to Vmap?

Reply:
Yes you are right, but JAX uses some lower-level optimization (e.x. memory access optimization) in most methods and perhaps `jax.vmap()` is one of them that's why people used that. So we should also use that ideally.

Reply:
I have made the changes suggested. Can you review it once?